compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -742,39 +742,23 @@ static LogicalResult setMatmulPeelingRootConfig(
     func::FuncOp entryPointFn, linalg::ContractionOpInterface op,
     ArrayRef<int64_t> distTileSizes, ArrayRef<int64_t> cacheTileSizes,
     ArrayRef<int64_t> vecTileSizes, int vectorSize) {
-  // The tiling for parallel dims and reduction dims should be separated.
-  SmallVector<int64_t> parallelTileSizes(vecTileSizes.begin(),
-                                         vecTileSizes.end());
-  parallelTileSizes.back() = 0;
-
-  // Clamp inner tiling sizes to avoid masking. The vector masking takes the
-  // last level of tiling to create masks. It would lead to incorrect masking if
-  // the inner tiling sizes are not clamped. Because padding won't be applied
-  // along those dimensions.
-  for (const auto &[index, size] : llvm::enumerate(distTileSizes)) {
-    if (!size)
-      continue;
-    parallelTileSizes[index] = std::min(parallelTileSizes[index], size);
-  }
-
-  // TODO(hanchung): Make logic more heuristic. Peeling hurts performance a lot
-  // if the dim size is small (e.g., K=24).
+  // The tiling for parallel dims and reduction dims should be separated, so we
+  // move K dim from parallel tile sizes to reduction tile sizes.
```

Comment:
[nit] Just to disambiguate. 
```suggestion
  // The tiling for parallel dims (M and N) and reduction dim (K) should be separated, so we
  // move K dim from parallel tile sizes to reduction tile sizes.
```

Reply:
thanks, this looks better!